system/lib/wasmfs/syscalls.cpp

```diff
@@ -327,6 +336,16 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
     // If O_DIRECTORY is also specified, still create a regular file:
     // https://man7.org/linux/man-pages/man2/open.2.html#BUGS
     if (flags & O_CREAT) {
+      // Since mode is optional, mode is specified using varargs.
+      mode_t mode = 0;
+      va_list vl;
+      va_start(vl, flags);
+      mode = va_arg(vl, int);
+      va_end(vl);
+      // Mask all permissions sent via mode.
+      // This is a placeholder for now which matches the JS filesystem.
+      mode &= S_IALLUGO;
```

Comment:
Why use `S_IALLUGO` here and `S_IRWXUGO` below? What's the difference? (As we discussed in chat, it would be nice to have our own more readable macros for permissions.)

Reply:
(Ignore this comment for this PR if this is entirely a change from #15445)

Reply:
Yep, these changes are from the previous PR.

Reply:
Can you do a merge (or if you have to, a rebase) to eliminate the unrelated changes from the diff on this PR?

Reply:
Is that possible since I merged the previous PR into this branch instead of main?